Improvement/59 Make board work with context freezing

NeillM opened the pull request:
As part of this I split the ability to post and rate into a new capability that is separate from the view capability.

I added the new capability into the checks for users being able to edit various features

A also made it so that comments also required the ability for users to make a comment in the first place.

An example of the admin user viewing a board in frozen context with these changes applied:
![image](https://user-images.githubusercontent.com/1038050/232450147-6762301c-a63a-41d4-80a2-717a73fdc1d6.png)

and the admin user viewing a comment in a frozen board:
![image](https://user-images.githubusercontent.com/1038050/232450671-cda521d6-c14c-4665-8fab-68bbf9111cd0.png)

NeillM commented:
This should resolve #59

NeillM commented:
I have rebased my changes for this on onto the MOODLE_400_STABLE branch as well

learningtechnologyservices commented on classes/board.php:
```diff
@@ -1264,10 +1264,17 @@ public static function board_is_editor($boardid) {
+
+        $context = static::context_for_board($boardid);
+        if (!has_capability('mod/board:post', $context)) {
+            // the user is not allowed to post via capabilities.
```
Sorry for this request, but CI checks are god, and they prefer comments starting with capital letters. TIA!

learningtechnologyservices commented on version.php:
```diff
@@ -25,7 +25,7 @@
 
 $plugin->component = 'mod_board'; // Full name of the plugin (used for diagnostics).
-$plugin->version   = 2022040105; // The current module version Use 2022.04.01 as base for 4.00.
+$plugin->version   = 2022040106; // The current module version Use 2022.04.01 as base for 4.00.
```
Version on main branch is now 2022040111, also release is '1.401.02 (Build 2022040111)';

Hoping to include this in the next release shortly, with some other PRs. TIA!

learningtechnologyservices reviewed (changes requested):
Hi Neill, many thanks for submitting this PR for improving Board! We'd like to include this in the next release, if we can get these couple of edits done by yourself please? TIA! Karen

P.S. We're just maintaining one active branch, MOODLE_401_STABLE, thanks!

NeillM commented:
I have rebased the changes (there was a merge conflict that was not part of the version file in the first commit)

I fixed the capitalisation on the comment as part of the rebase.

danowar2k commented:
Eagerly awaiting the split of post and view capabilities :-)

NeillM commented:
Rebased the change
